test/integration/endpointslice/endpointslicemirroring_test.go

```diff
@@ -228,3 +231,199 @@ func TestEndpointSliceMirroring(t *testing.T) {
 	}
 
 }
+
+func TestEndpointSliceMirroringUpdates(t *testing.T) {
+	masterConfig := framework.NewIntegrationTestMasterConfig()
+	_, server, closeFn := framework.RunAMaster(masterConfig)
+	defer closeFn()
+
+	config := restclient.Config{Host: server.URL}
+	client, err := clientset.NewForConfig(&config)
+	if err != nil {
+		t.Fatalf("Error creating clientset: %v", err)
+	}
+
+	resyncPeriod := 12 * time.Hour
+	informers := informers.NewSharedInformerFactory(client, resyncPeriod)
+
+	epsmController := endpointslicemirroring.NewController(
+		informers.Core().V1().Endpoints(),
+		informers.Discovery().V1beta1().EndpointSlices(),
+		informers.Core().V1().Services(),
+		int32(100),
+		client,
+		1*time.Second)
+
+	// Start informer and controllers
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	informers.Start(stopCh)
+	go epsmController.Run(1, stopCh)
+
+	testCases := []struct {
+		testName      string
+		tweakEndpoint func(ep *corev1.Endpoints)
+	}{
+		{
+			testName: "Update labels",
+			tweakEndpoint: func(ep *corev1.Endpoints) {
+				ep.Labels["foo"] = "bar"
+			},
+		},
+		{
+			testName: "Update annotations",
+			tweakEndpoint: func(ep *corev1.Endpoints) {
+				ep.Annotations["foo2"] = "bar2"
```

Comment:
The tests don't assert on the value of the annotations.  If you change this to just the following line:

```go
ep.Labels["foo"] = "bar"
```

then the update will succeed but we never look at the annotation values that should be propagated.

For example:

```diff
diff --git a/test/integration/endpointslice/endpointslicemirroring_test.go b/test/integration/endpointslice/endpointslicemirroring_test.go
index 409e60a7368..e0ad8fbf260 100644
--- a/test/integration/endpointslice/endpointslicemirroring_test.go
+++ b/test/integration/endpointslice/endpointslicemirroring_test.go
@@ -273,7 +273,8 @@ func TestEndpointSliceMirroringUpdates(t *testing.T) {
                {
                        testName: "Update annotations",
                        tweakEndpoint: func(ep *corev1.Endpoints) {
-                               ep.Annotations["foo2"] = "bar2"
+                               // ep.Annotations["foo2"] = "bar2"
+                               ep.Labels["foo"] = "bar"
                        },
                },
                {
```

```sh
$ pwd
/Users/amcdermo/go-projects/kube/src/k8s.io/kubernetes/test/integration/endpointslice
$ go test
...
PASS
I0118 11:08:58.852269   80756 etcd.go:140] etcd exit status: signal: killed
ok  	k8s.io/kubernetes/test/integration/endpointslice	21.721s
$ echo $? 
0

Reply:
we´ve talked offline, the test logic is testing behaviour independently of the inputs.
So the test_cases inputs are the ones that defines what is being tested.